### ISSUE TITLE
Copy and paste doesn't work from server to client

### ISSUE THREAD

**Issue by chadaaland** [symless/synergy#5502]
Server: Mac osX 10.11.5
Client: Windows 10 
Synergy Version: 1.7.6

Copy and paste does work when I open/show the synergy application on the client, then go back and copy, then go back and paste, but otherwise doesn't work.

**Comment by LittleLama** [comment]
Same issue between two windows 7 
Server x64 v1.8.2 beta 
Client x86 v1.7.6

Thanks for the tip.

**Comment by XinyuHou** [comment]
@chadaaland @LittleLama 
Could you send us some logging using gist?
https://gist.github.com/

**Comment by chadaaland** [comment]
I am covering for Chad the next 3 weeks.  Can you please explain more what you need and I will try to help.

Thanks!


Katie Potter

Vice President 

Move Management Group Inc.
372 S. Eagle Road #154
Eagle, ID 83616-5908
877-664-4968 x102
877-381-1754 Fax



Visual/Telephone/Video Surveys ~ Debris Pick-Ups

QC Checks ~ Transit Damage Reviews ~ Mover Training DVDs

**Comment by ngoldbaum** [comment]
I'm having the same issue. One other piece of information I discovered while trying to generate the logs you asked for: it doesn't seem to be an issue if both client and server have the logging level set to debug.

In this example I start synergy on both the client and the server, copy some text on the server, then mouse over to the client.

Here's the debug log from the OSX server: https://gist.github.com/ngoldbaum/76a41523e485a9a41f087aec965c4351

And here's the debug log from the Windows client: https://gist.github.com/ngoldbaum/cfa2ae9f1d81dbb7a849a054608ed892

Here the server's logging level is set to info so I can trigger the issue.

**Comment by ngoldbaum** [comment]
@XinyuHou @chadaaland in case you didn't see the comment I left yesterday ^

**Comment by nlyan** [comment]
I can't seem to reproduce this guys. Our Mac OS X 10.11.06 server and Windows 10 client config is working well here for bidirectional clipboard support, with both set to the "Warning" log level.

**Comment by LittleLama** [comment]
Here some logs

<script src="https://gist.github.com/LittleLama/5e21cd5ae0518c22bf9b8bf9ae9b745d.js"></script>

<script src="https://gist.github.com/LittleLama/2369dee4c346536761bc9940f436ca06.js"></script>

**Comment by slagelwa** [comment]
Confirming, same issue here too.  

Client: Mac OSX 10.10.5
Server: Windows 7
Synergy: 1.8.2-stable-36cd521

**Comment by ngoldbaum** [comment]
Just updated synergy to 1.8.2 - still the same issue, can't copy/paste from a mac server to a windows client.

**Comment by mal** [comment]
Same issue, didn't have it prior to 1.8.2.

Client: Ubuntu 16.04
Server: Windows 10
Synergy: 1.8.2-stable-36cd521
Logging Level: info

**Comment by vanowm** [comment]
v1.8.3-beta-fa9f30f (W10 x64 server, W7 x64 client) also suffers from copy to clipboard issue
Very often copy text on server prevents clipboard from populating on server itself (no paste is possible)!!!
When that happens, switching to client produces error on client side:
> [2016-08-13T10:26:58] ERROR: corrupted clipboard data, expected size=19 actual size=0
The solution is to select different text and copy it. Repeat until it works.

**Comment by nlyan** [comment]
We have identified the probable cause of this.

For those not receiving "corrupted clipboard data" messages, you should be able to verify that if you wait a second or two after copying _but before moving your cursor to another screen_ that paste works more reliably on the receiving side.

**Comment by mal** [comment]
Any news for those that do see `corrupted clipboard data ... size=0`? Is it a separate issue? For now I've just rolled back to `1.7.6` :frowning:

**Comment by XinyuHou** [comment]
> After upgrading to 1.8.2, more often than not I'm unable to copy something from the synergy client on my Mac (OSX 10.11) to my server on Ubuntu 14.04.4 (64-bit).
> 
> This is not 100% consistent, but in most cases it seems that copying formatted text (e.g. from a webpage) works, but copying something from a terminal, Chrome's URL bar, etc. doesn't.
> 
> I'm attaching DEBUG1 logs from client and server for a few attempts - some successful, some not. I've taken the liberty to replace my hostnames with MYMACHINENAME and MYOTHERMACHINENAME and to change my home directory to something that doesn't give away my company's name or computer setup.
> 
> I'll take a wild guess that this is the relevant message:
> [2016-08-11T13:13:06] DEBUG1: clipboard: sending failure to 0x00c00007,31,0
> Let me know if you need DEBUG2 (or if you want to send me custom binaries with more logging).
> 
> [synergy.zip](https://github.com/symless/synergy/files/414148/synergy.zip)

**Comment by XinyuHou** [comment]
> After upgrade to the latest stable build, I am no longer able to copy images from Win10 to Win10 using the x64 client, Frequently (if not persistently) the file errors in the logs with teh following
> 
> [2016-08-11T11:55:50] ERROR: corrupted clipboard data, expected size=1334452 actual size=69632
> [2016-08-11T11:55:50] ERROR: corrupted clipboard data, expected size=1334452 actual size=0
> 
> This worked for a long time... although sometimes broke with the locked symlink (which this build is supposed to fix).
> 
> I searched issues and did not see it posted yet. Wondering if anyone else is seeing this or if it is a "me centric" issue.
> 
> Version 1.8.2-stable-36cd521

**Comment by XinyuHou** [comment]
> Wow.
> 
> I'd previously missed that you we were talking specifically about images. To test I took a screenshot on the server (Windows 10, Print Scrn) then moved the mouse over to the client (Ubuntu 16.04). The mouse flickered for a very short time, then centered itself on the client (as the client stoped/crashed).
> 
> The mouse pointer never visibly returned to the server, instead the CPU fan went nuts and all input was blocked. Ctrl+Alt+Delete was the only thing to function (not even Ctrl+Shift+Esc), from there I was able to start task manager (input appeared fine on the CAD menu) but could not interact with it as input was still hijacked, so ulimately had to use the CAD menu to force a reboot.
> 
> Obviously the server log was lost in the process, however I was able to retrieve the client log from the syslog (FYI that's a manual restart at 12:17):
> 
> Aug 14 12:06:45 hostname synergyc[13824]: [2016-08-14T12:06:45] INFO: entering screen
> Aug 14 12:06:58 hostname synergyc[13824]: [2016-08-14T12:06:58] INFO: leaving screen
> Aug 14 12:14:36 hostname synergyc[13824]: [2016-08-14T12:14:36] INFO: entering screen
> Aug 14 12:14:36 hostname synergyc[13824]: [2016-08-14T12:14:36] INFO: clipboard was updated
> Aug 14 12:14:36 hostname synergyc[13824]: [2016-08-14T12:14:36] INFO: clipboard was updated
> Aug 14 12:14:37 hostname synergyc[13824]: [2016-08-14T12:14:37] INFO: leaving screen
> Aug 14 12:14:42 hostname synergyc[13824]: [2016-08-14T12:14:42] INFO: entering screen
> Aug 14 12:14:42 hostname synergyc[13824]: [2016-08-14T12:14:42] INFO: leaving screen
> Aug 14 12:14:49 hostname synergyc[13824]: [2016-08-14T12:14:49] NOTE: server is dead
> Aug 14 12:14:50 hostname synergyc[13824]: [2016-08-14T12:14:50] WARNING: failed to connect to server: server is not responding
> Aug 14 12:14:51 hostname synergyc[13824]: [2016-08-14T12:14:51] NOTE: connecting to 'xxx.xxx.xxx.xxx': xxx.xxx.xxx.xxx:24800
> Aug 14 12:14:51 hostname synergyc[13824]: [2016-08-14T12:14:51] INFO: OpenSSL 1.0.2g-fips  1 Mar 2016
> Aug 14 12:15:02 hostname synergyc[13824]: [2016-08-14T12:15:02] ERROR: failed to connect secure socket
> Aug 14 12:15:03 hostname synergyc[13824]: [2016-08-14T12:15:03] NOTE: disconnected from server
> Aug 14 12:15:03 hostname synergyc[13824]: [2016-08-14T12:15:03] NOTE: stopped client
> Aug 14 12:17:02 hostname systemd[1]: Starting Synergy client service...
> Aug 14 12:17:02 hostname systemd[1]: Started Synergy client service.
> Aug 14 12:17:02 hostname synergyc[26878]: [2016-08-14T12:17:02] NOTE: started client
> Won't be trying to copy any images again soon :fearful:

**Comment by XinyuHou** [comment]
> For me it is not just images. Text too. I dont think my server is crashing either. Restarting the client has been bringing it back... but it seems I get 1ish copy and paste before it breaks. The server does get teh clipboard copied too because I can paste the content locally.

**Comment by vmtyler** [comment]
I've been having the same issue with OS X 10.11 -> OS X 10.11 and @nlyan's note was easily reproducible- if you move to the other screen too fast after copy, it doesn't work.

**Comment by ekondrashev** [comment]
Same for me. 
Server OSX 10.11
Client Ubuntu 15.10

Copying from client works, but copy from Server - not

**Comment by nova77** [comment]
I am not sure if it's related but I am having clipboard issues (about 30% of the times it does not capture) where the server is Ubuntu and Client is OSX, and only appeared since my last upgrade to 1.8.2.

**Comment by mrog** [comment]
Same issue here.
Synergy: 1.8.2
Server: Ubuntu 16.04
Client: OSX 10.11.6

I can copy text from Ubuntu to Mac 100% of the time, but the other direction usually fails.

**Comment by staaronis3** [comment]
The copy-paste doesn't work in my case when I am connected to a computer via Teamviewer.

When I stop the synergy server and/or close the teamviewer the functionality of copy-paste goes back to normal.

I haven't the ability to change the elevated mode to see if it can fix the problems, as I have mentioned in #5583.

Server: Windows 10 x64, Client: Windows 7 x64
Both with the 1.8.3-rc1-e36a7ae version of Synergy.

**Comment by raymundocvela-nimax** [comment]
Same for me, I can copy from de Server and paste in the client, but I can´t copy from client and paste in client neiter  to server.

**Comment by slagelwa** [comment]
If it's any help, it seems to copy&paste seems to work (windows server, macosx client) if you don't have any Windows active on the server, move the mouse to the client, copy, and move the mouse back over without activating a window.  They activating a window and hitting paste.

**Comment by LittleLama** [comment]
Hi.  Thanks for your feedback,  but read the title, we're trying to do the
opposite

**Comment by slagelwa** [comment]
Ok, so is there an open issue about copying from a macosx client to a windows server not using?

**Comment by XinyuHou** [comment]
hi guys, could you test this nightly please?

http://symless.com/nightly?filter=b55f64

**Comment by LeadManPL** [comment]
Testing Bug #5502

SRV: synergy-v1.8.3-rc2-89c9c58-Linux-x86_64
CLN: synergy-v1.8.3-rc2-89c9c58-Windows-x64

Thes steps will test both #4740 and #5502. Because the bugs happen intermittently, this requires up to a day of testing, sorry about that.
1. Copy and paste anything less than 5MB between machines
2. See that content has been pasted without causing a disconnect
3. Continue testing until your eyes begin to bleed
 
Copied various plain texts both directions. Copied images. Synergy was stable.
One time, while copying 744kB of plain text from synergy log, from server to client, after presing ^V on client in Microsoft Word application, there appeared tons of asian signs instead of plain text. During paste operation, I was switching screens rapidly. Synergy stayed connected after pasting Chinese characters.  It occurred only once. Next several attempts to reproduce Asian signs failed. See attached screen shot.

PASSED. Synergy worked stable, apart from one occurrence of pasting Chinese signs.

**Comment by nlyan** [comment]
@LeadManPL If those 'Chinese signs' or any other artifacts appear again can you look at the tail end of your logs for anything conspicuous? You may have to run Synergy at the DEBUG1 log level.

**Comment by LeadManPL** [comment]
@nlyan Lucky you! :) 
I just got this error again.
Log files and screen shots attached.
[debug.zip](https://github.com/symless/synergy/files/490111/debug.zip)

**Comment by crunchy234** [comment]
I have the same issue when copying from windows (server) to Ubuntu (client) is there a build I should try? Note: The my current version of synergy is: ```1.8.7-stable-9799e96``` on both client and server.
Copying from Ubuntu (client) to Windows (server) works perfectly.

**Comment by CMCDragonkai** [comment]
Would clipboard copy be affected by different versions on the client side and server side? My linux client is on an older version while my windows server is on the latest version.

**Comment by vanowm** [comment]
@CMCDragonkai: doesn't look like it, . However running different OS could affect, at least in my case server - W10, client - W7
Although I haven't experienced this issue for a very long time now.

**Comment by CMCDragonkai** [comment]
Well clipboard copy from NixOS client and Windows server isn't working atm.

**Comment by fglez-tsl** [comment]
I have the same issue when copying from windows-7 (server) to Ubuntu 12.04 (client). I am using 1.8.7-stable-9799e96 on both client and server and copying from client to server works ok.

**Comment by NicoGrapsas** [comment]
Same issue here.
Copying from Windows client to Windows server doesn't work but i can copy from client to server.
The problem started suddenly because it was working ok till now. I didn't update.

**Comment by crunchy234** [comment]
I have found that I can not copy from windows client to fedora server does not work when both machines startup. However when I open Synergy on the windows client and hit apply (so it restarts the client) copying and pasting from the windows client works until I need to reboot the windows client.

I also had to change my Elevate setting from 'Always' to 'As Needed' to get this to work.

**Comment by cavagnaro** [comment]
Why is this closed? What is the solution? I have the same exact problem

**Comment by LeadManPL** [comment]
I have tested and It was fine in 2016. Unfortunately in recent builds, occasionally I also loose ability to copy between machines, thus I think this Issue should be reopened...

**Comment by cavagnaro** [comment]
Agree, I have the same problem between Windows 10 computers. Copy/Paste doesn't work at all

**Comment by anhar** [comment]
I'm having the exact same problem:

**Server:** OSX 10.12.4, 1.8.8-stable-25a8cb2
**Client:** Windows 10 Home 64-bit (10.0, Build 15063), 1.8.8-stable-25a8cb2
**Synergy log level debug2:** [log gist (removed some "mouse move" events](https://gist.github.com/anhar/0364d1802212065f35a549d3248d325c)

**Comment by sshling** [comment]
restart the client  ,resovle all ....
